Composer/packages/client/src/pages/botProject/adapters/ABSChannels.tsx

```diff
@@ -162,26 +125,18 @@ export const ABSChannels: React.FC<RuntimeSettingsProps> = (props) => {
       if (profile) {
         const config = JSON.parse(profile.configuration);
 
-        let newtoken = '';
-        if (userShouldProvideTokens()) {
-          if (isShowAuthDialog(false)) {
-            setShowAuthDialog(true);
-          }
-          newtoken = getTokenFromCache('accessToken');
-        } else {
-          newtoken = await getTokenInteractively(config.tenantId);
-        }
-        setToken(newtoken);
-
-        if (newtoken) {
-          setCurrentResource({
-            microsoftAppId: config?.settings?.MicrosoftAppId,
-            resourceName: config.botName || config.name,
-            resourceGroupName: config.resourceGroup || config.botName || config.name,
-            tenantId: config.tenantId,
-            subscriptionId: config.subscriptionId,
-          });
-        }
+        // NOTE: if config.tenantId is missing (as might be the case with a pre-1.4 composer publish profile)
+        // this will cause the user to be prompted for the correct tenant when they login.
+        // if they choose the WRONG tenant, an error will appear.
+        requireUserLogin(config.tenantId);
+
+        setCurrentResource({
+          microsoftAppId: config?.settings?.MicrosoftAppId,
+          resourceName: config.botName || config.name,
```

Comment:
if config can be undefined (Line 134 thinks so) this and the lines below with config.something will throw

Reply:
it should be set. unless the json parse failed, which shouldn't happen, but I added a try/catch just in case.